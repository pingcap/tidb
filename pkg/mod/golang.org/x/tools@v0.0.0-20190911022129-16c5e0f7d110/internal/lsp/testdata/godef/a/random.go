package a

func Random() int { //@Random
	y := 6 + 7
	return y
}

func Random2(y int) int { //@Random2,mark(RandomParamY, "y")
	return y //@godef("y", RandomParamY)
}

type Pos struct {
	x, y int //@mark(PosX, "x"),mark(PosY, "y")
}

func (p *Pos) Sum() int { //@mark(PosSum, "Sum")
	return p.x + p.y //@godef("x", PosX)
}

func _() {
	var p Pos
	_ = p.Sum() //@godef("()", PosSum)
}
