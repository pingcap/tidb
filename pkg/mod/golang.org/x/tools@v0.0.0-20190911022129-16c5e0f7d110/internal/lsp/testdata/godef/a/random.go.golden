-- PosSum-definition --
godef/a/random.go:16:15-18: defined here as func (*Pos).Sum() int

-- PosSum-definition-json --
{
	"span": {
		"uri": "file://godef/a/random.go",
		"start": {
			"line": 16,
			"column": 15,
			"offset": 248
		},
		"end": {
			"line": 16,
			"column": 18,
			"offset": 251
		}
	},
	"description": "func (*Pos).Sum() int"
}

-- PosSum-hover --
func (*Pos).Sum() int
-- PosX-definition --
godef/a/random.go:13:2-3: defined here as field x int

-- PosX-definition-json --
{
	"span": {
		"uri": "file://godef/a/random.go",
		"start": {
			"line": 13,
			"column": 2,
			"offset": 187
		},
		"end": {
			"line": 13,
			"column": 3,
			"offset": 188
		}
	},
	"description": "field x int"
}

-- PosX-hover --
@mark(PosX, "x"),mark(PosY, "y")
field x int
-- RandomParamY-definition --
godef/a/random.go:8:14-15: defined here as var y int

-- RandomParamY-definition-json --
{
	"span": {
		"uri": "file://godef/a/random.go",
		"start": {
			"line": 8,
			"column": 14,
			"offset": 79
		},
		"end": {
			"line": 8,
			"column": 15,
			"offset": 80
		}
	},
	"description": "var y int"
}

-- RandomParamY-hover --
var y int
